app/Http/Controllers/ApiPhotosController.php

```diff
@@ -175,6 +176,14 @@ public function store (Request $request) :array
             \Log::info(['ApiPhotosController@store', $e->getMessage()]);
         }
 
+        // Since a user can upload multiple photos at once,
+        // we might get old values for xp, so we update the values directly
+        // without retrieving them
+        $user->update([
+            'xp' => DB::raw('xp + 1'),
+            'total_images' => DB::raw('total_images + 1')
+        ]);
```

Comment:
We will migrate this to Redis soon which should fix this issue

Reply:
Upgrading this to Redis might take some time I think, even if we prioritize it. I'd like to do the migration using tests, I feel it's going to introduce a lot of new bugs otherwise. For that reason, this update() operation fixes the issue for now.